dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -221,100 +140,29 @@ class DataManager(
-                        Argument(name = "defaultRequeueRejected", value = "false"),
-                    ],
-                ),
-                exchange = Exchange(ExchangeName.ItemStored, declare = "false"),
-                key = [RoutingKeyNames.data],
-            ),
-        ],
-    )
-    fun removeStoredItemFromTemporaryStore(
-        @Payload dataId: String,
-        @Header(MessageHeaderKey.CorrelationId) correlationId: String,
-        @Header(MessageHeaderKey.Type) type: String,
-    ) {
-        messageUtils.validateMessageType(type, MessageType.DataStored)
-        if (dataId.isEmpty()) {
-            throw MessageQueueRejectException("Provided data ID is empty")
-        }
-        logger.info("Internal Storage sent a message - job done")
-        logger.info(
-            "Dataset with dataId $dataId was successfully stored. Correlation ID: $correlationId.",
-        )
-        messageUtils.rejectMessageOnException {
-            publicDataInMemoryStorage.remove(dataId)
-        }
-    }
-
     /**
      * Method to make the data manager get the data of a single entry from the data store
      * @param dataId to identify the stored data
      * @param dataType to check the correctness of the type of the retrieved data
      * @param correlationId to use in combination with dataId to retrieve data and assert type
      * @return data set associated with the data ID provided in the input
      */
-    fun getDataSet(dataId: String, dataType: DataType, correlationId: String): StorableDataSet {
-        assertActualAndExpectedDataTypeForIdMatch(dataId, dataType, correlationId)
-        val dataMetaInformation = metaDataManager.getDataMetaInformationByDataId(dataId)
-        lateinit var dataAsString: String
-        try {
-            dataAsString = getDataFromCacheOrStorageService(dataId, correlationId)
-        } catch (e: ClientException) {
-            handleInternalStorageClientException(e, dataId, correlationId)
-        }
-        logger.info("Received Dataset of length ${dataAsString.length}. Correlation ID: $correlationId")
-        val dataAsStorableDataSet = objectMapper.readValue(dataAsString, StorableDataSet::class.java)
-        dataAsStorableDataSet.requireConsistencyWith(dataMetaInformation)
-        return dataAsStorableDataSet
-    }
-
-    private fun handleInternalStorageClientException(e: ClientException, dataId: String, correlationId: String) {
-        if (e.statusCode == HttpStatus.NOT_FOUND.value()) {
-            logger.info("Dataset with id $dataId could not be found. Correlation ID: $correlationId")
-            throw ResourceNotFoundApiException(
-                "Dataset not found",
-                "No dataset with the id: $dataId could be found in the data store.",
-                e,
-            )
-        } else {
-            throw e
-        }
+    fun getPublicDataSet(dataId: String, dataType: DataType, correlationId: String): StorableDataSet {
```

Comment:
FInde es verwirrend, dass es `getPublicDataSet()` gibt, aber auch `getPublicData()` und dann nochmal `getDataFromCacheOrStorageService()`
Das sollten wir einfacher gestalten denke ich. SCheint mir aktuell zu kompliziert.

Reply:
Aber das sind strukturell die 3 Funktionen, die wir auch bisher haben. Aber lass uns gerne mal über den Aufbau sprechen

Reply:
Ja gerne. Der ist auch schon in main iwie komisch.

Reply:
Ich mache einen VOrschlag und melde mich